app/frontend/wallet/shelley/transaction/utils.ts

```diff
@@ -1,53 +1,35 @@
 import BigNumber from 'bignumber.js'
-import distinct from '../../../helpers/distinct'
 import {Address, CertificateType, Lovelace, TokenBundle} from '../../../types'
-import {aggregateTokenBundles} from '../../helpers/tokenFormater'
+import {encodeCbor} from '../../helpers/cbor'
 import {TxCertificate, TxPlanAuxiliaryData, TxInput, TxOutput, TxWithdrawal} from '../../types'
-import {MIN_UTXO_VALUE} from './constants'
+import {cborizeSingleTxOutput} from '../shelley-transaction'
 import {estimateTxSize} from './estimateTxSize'
 
-export const computeMinUTxOLovelaceAmount = (tokenBundle: TokenBundle): Lovelace => {
-  // based on https://github.com/input-output-hk/cardano-ledger-specs/blob/master/doc/explanations/min-utxo.rst
-  const quot = (x: number, y: number) => Math.floor(x / y)
-  const roundupBytesToWords = (x: number) => quot(x + 7, 8)
-  // TODO: this to network config or constants
-  const minUTxOValue = new BigNumber(MIN_UTXO_VALUE)
-  // NOTE: should be 2, but a bug in Haskell set this to 0
-  const coinSize = 0
-  const txOutLenNoVal = 14
-  const txInLen = 7
-  const utxoEntrySizeWithoutVal = 6 + txOutLenNoVal + txInLen // 27
+export const computeMinUTxOLovelaceAmount = (
+  address: Address,
+  tokenBundle: TokenBundle
+): Lovelace => {
+  // Reference:
+  // - https://cips.cardano.org/cips/cip55/#thenewminimumlovelacecalculation
+  // - https://hydra.iohk.io/build/15339994/download/1/babbage-changes.pdf page 9
+  const overhead = 160
 
-  // NOTE: should be 29 but a bug in Haskell set this to 27
-  const adaOnlyUtxoSize = utxoEntrySizeWithoutVal + coinSize
+  // taken from https://cexplorer.io/params
+  const coinsPerUTxOByte = 4310
 
-  // this ensures there are only distinct (policyId, assetName) pairs
-  const aggregatedTokenBundle = aggregateTokenBundles([tokenBundle])
+  // 80kb is current max block size
+  const coinsUpperBound = new BigNumber(80 * 1024 * coinsPerUTxOByte) as Lovelace
```

Comment:
It's not clear to me why `coinsUpperBound` depends on max block size. Could you please explain? I didn't find any explanation in the referenced docs.

Reply:
my understanding is that we want to use the max possible amount of coins as a placeholder and I don't see a relation with the block size either. In the CDDL the type of "coin" is "uint" which I didn't find a direct definition for, but I'm pretty sure it's roughly 2^64-1 which is also the limit HW wallets work with